### ISSUE TITLE
Better Couchpotato Migration

### ISSUE THREAD

**Issue by Th3MadHatter** [Radarr/Radarr#3187]
**Is your feature request related to a problem? Please describe.**
Couchpotato is ancient but it still works and a lot (like me) of people migrated recently or going to migrate.
The problem is that( (i'm going to give a stupid example) things like The Tiny World (1600) are added to the library perfectly but Tiny World, The (1600) are often not added at all, manually adding them works just fine even pointing to Tiny World, The (1600) but it doesn't add it automatically.
This is completely useless for anything other then Couchpotato lists but still something very useful for migration.

**Describe the solution you'd like**
Make couchpotato lists check for both The Tiny World (1600), and Tiny World, The (1600) problem solved.

**Describe alternatives you've considered**
Check for multiple forms of index data based on older or different applications to avoid ignoring them.

**Additional context**
None

**Comment by fryfrog** [comment]
Are you talking about *existing* files? Or the ability to import wanted items from CP? Radarr happily supports `Movie Title, The (Year)` format itself, so I'd be a little surprised if that doesn't work during bulk import or any of the other imports. But maybe it doesn't work during import of wanted items?

**Comment by galli-leo** [comment]
If you have existing files, do not use the couchpotato list. It will not correctly take the folder from CP. Use bulkimport for that. The CP list is only for not already downloaded movies.

**Comment by Th3MadHatter** [comment]
I tried both the bulk import and the list option, I still have lots of directories that I have to add manually.
To give an example couchpotato directory structures of my Hobbit Blu-ray's don't work unless I manually point to the right directory, this is the same with lots of other titles but I happen to have the Blu-ray laying next to me and I know I had issues getting it added.

**Comment by fryfrog** [comment]
Radarr needs a flat `folder/file.ext` structure. For example, you'd have `/Movies/Movie Title (Year)/Movie Title (Year) Edition [Quality].ext`. When you bulk import in Radarr, you point at `/Movies` and it all works.

I'm guessing from your description that you probably have a `/Movies/The Hobbit/The Hobbit Part 1 (2011)/The Hobbit Part 1 (2011) Bluray.ext` and you're pointing at `/Movies` and it isn't being found. You'd either need to flatten it out *or* import that as a root folder. And if you're doing that for The Hobbit, you're doing it for a bunch of others. I would not recommend this setup.

You should get on Discord, I don't think your problem is a feature request or bug. I think you're just not doing Radarr right. It'll be easier in real time on the support channel. Be sure to link to your issue to give a little context.

**Comment by Th3MadHatter** [comment]
Not at all, let me give you the exact folder structure.

X:\Movie\Hobbit The Desolation of Smaug, The (2013)\The Hobbit The Desolation of Smaug.mkv'

Which should be picked up but it simply doesn't, most likely because couchpotato just renamed it very weirdly but Kodi has no issues with it nor does Plex. 

Can't get on discord on here else I would have, kind of taking a break from work lol, github is common to visit and even IRC is but discord isn't :stuck_out_tongue_closed_eyes:

**Comment by fryfrog** [comment]
Dang, that one should totally work. I'd put that one folder into another empty folder, then turn logging up to trace and do a bulk import on *just* that empty folder. See what you see in the logs. The shuffling things around is so the logs will be simpler.

**Comment by fryfrog** [comment]
Wait, your example has the year in the *folder*, but is it really missing from the *file* name? If so, add year to the file name.

**Comment by Th3MadHatter** [comment]
It is missing the year in the file, that's the problem with a lot of long file names that where over complicated in couchpotato, they removed the year from the file and just changed it to the original name,... or maybe radarr did that, I manually added it already so it's not the best example, I'll try and go through my list and see if I can find something that doesn't add as well but it's all similar to this kind of structure.

**Comment by Th3MadHatter** [comment]
@fryfrog I found a couple of examples, this is really the way couchpotato renamed things so adding a rule to add these would be excellent for people migrating. The name of the movie itself is missing the year, you could say add it manually and that is fine for people with a library of 10 files but go do that with a 1000+ :rofl: 

\Movie\First Purge, The (2018)\The First Purge.mkv
\Movie\Matrix Reloaded, The (2003)\The Matrix Reloaded.mkv
\Movie\Matrix Revolutions, The (2003)\The Matrix Revolutions.mkv
\Movie\40 Year Old Virgin, The (2005)\The 40 Year Old Virgin.mkv

You get the idea I hope :smile:

**Comment by Th3MadHatter** [comment]
My CP template was default on the master windows edition branch... I've never changed it in all the years I used it.
It is no longer running or installed on any device.

I never liked the naming either but Kodi and Plex pick it up so I never bothered changing it, CP worked but unfortunately the project died and I had to migrate.

I don't see why radarr didn't add them either but for some reason it doesn't. Since this is not a homebrew template there might be more people migrating in the future because most people simply don't know that CP died. It might be useful to add support for the template in multi format of directory alone, nfo file if available or file name and then search for possible corresponding years.

I don't ask this for myself, I can fix this just fine, even manually, but there will be a lot of people that can't, I know a lot of people that can't work with filebot if their life depended on it and can barely use setups as is so, to help those people migrating quickly and without too many issues or bug reports I'm asking if you want to take a look at the script making it more adoptable for various directory structures. I know it's not extra work you're waiting for and I would have done it but I'm not even at full understanding of how every function in Radarr works yet, let alone ready to tinker with the code and make PR's.

Thanks :smile:
